packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/main/java/com/example/alternate_language_test_plugin/AlternateLanguageTestPlugin.java

```diff
@@ -146,6 +146,24 @@ public void echoAsyncString(@NonNull String aString, Result<String> result) {
     result.success(aString);
   }
 
+  @Override
+  public void throwAsyncError(Result<Object> result) {
+    try {
+      throw new RuntimeException("An error");
+    } catch (Exception e) {
+      result.error(e);
+    }
```

Comment:
I'm not a Java expert, but couldn't you just collapse all of this to `result.error(new RuntimeException("An error"))`? I wouldn't think the try/throw/catch would be necessary.